themes/classic/templates/catalog/_partials/product-details.tpl

```diff
@@ -30,7 +30,8 @@
     {if $product.show_quantities}
       <div class="product-quantities">
         <label class="label">{l s='In stock' d='Shop.Theme.Catalog'}</label>
-        <span>{$product.quantity} {$product.quantity_label}</span>
+        <span data-stock="{$product.quantity}">{$product.quantity} {$product.quantity_label}</span>
+        <input type="hidden" id="allow_oosp" value="{$product.allow_oosp}">
```

Comment:
Is this input part of a form?

If it is, then you shouldn't trust it on the receiving end, as it can be manipulated by the client.

If it's not (and if you are using it only to store information for the javascript code), then it would probably be best to store it as a data attribute somewhere else.